internal/provider/resource_gitlab_group_project_file_template_test.go

```diff
@@ -0,0 +1,149 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+	"os"
+	"strconv"
+	"testing"
+)
+
+func TestAccGitlabGroupProjectFileTemplate_basic(t *testing.T) {
+	// Since we do some manual setup in this test, we need to handle the test skip first.
+	if os.Getenv(resource.EnvTfAcc) == "" {
+		t.Skip(fmt.Sprintf("Acceptance tests skipped unless env '%s' set", resource.EnvTfAcc))
+	}
+	baseGroup, firstProject, secondProject := setupTestData(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckProjectFileTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				SkipFunc: isRunningInCE,
+				Config:   testAccGroupProjectFileTemplateConfig(baseGroup.ID, firstProject.ID),
+				Check: resource.ComposeTestCheckFunc(
+					// Note - we can't use the testAccCheckGitlabGroupAttributes, because that checks the TF
+					// state attributes, and file project template explicitly doesn't exist there.
+					testAccCheckGitlabGroupFileTemplateValue(baseGroup, firstProject),
+					testAccCheckGitlabGroupFileTemplateStateValue(baseGroup, firstProject),
+				),
+			},
+			{
+				//Test that when we update the project name, it re-links the group to the new project
+				SkipFunc: isRunningInCE,
+				Config:   testAccGroupProjectFileTemplateConfig(baseGroup.ID, secondProject.ID),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabGroupFileTemplateValue(baseGroup, secondProject),
+					testAccCheckGitlabGroupFileTemplateStateValue(baseGroup, secondProject),
+				),
+			},
+			{
+				//Test that when we remove the resource, the group gets its value set to 0
+				SkipFunc: isRunningInCE,
+				Config:   testAccGroupProjectFileTemplateConfig(baseGroup.ID, secondProject.ID),
+				Destroy:  true,
+			},
+		},
+	},
+	)
+}
+
+func testAccCheckGitlabGroupFileTemplateStateValue(group *gitlab.Group, project *gitlab.Project) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		groupID := state.RootModule().Resources["gitlab_group_project_file_template.linking_template"].Primary.Attributes["group_id"]
+		projectID := state.RootModule().Resources["gitlab_group_project_file_template.linking_template"].Primary.Attributes["file_template_project_id"]
+
+		if groupID != strconv.Itoa(group.ID) {
+			return fmt.Errorf("Group didn't match with the state file. State had %s and expected %d", groupID, group.ID)
+		}
+
+		if projectID != strconv.Itoa(project.ID) {
+			return fmt.Errorf("Project didn't match with the state file. State had %s and expected %d", projectID, project.ID)
+		}
+		return nil
+	}
+}
+
+func setupTestData(t *testing.T) (*gitlab.Group, *gitlab.Project, *gitlab.Project) {
```

Comment:
Why not use the test helpers from `helper_test.go`? You may need to create a new one to create a project within a group.

Reply:
Updated - I created a new method to create a project with a namespace when the input namespace != 0, and refactored the existing createProject to call that with a 0 namespace so we have less duplicated code.